api/proto/profile/local_profile.proto

```diff
@@ -69,3 +69,52 @@ message LocalAppInfo {
    org.lfedge.eve.info.ZSwState state = 5;
 }
 
+// LocalAppCmds message is sent by the Local Profile Server in response to a GET
+// request issued by EVE to the api/v1/app_commands API.
+message LocalAppCmds {
+   // Security token. EVE will verify that server_token matches the profile server
+   // token received from the controller.
+   string server_token = 1;
+   // A list of commands requested to execute for locally running application instances.
+   // A new request created for the same application and the same command should overwrite
+   // the previous entry with the 'timestamp' field updated.
+   // It is not required for the Local profile server to persist command requests.
+   // Instead, it is expected that after a restart of the Local profile server,
+   // the user would repeat the request if the previous one failed or didn't yield
+   // the desired outcome.
+   repeated AppCommand app_commands = 2;
+}
+
+// AppCommand references a running application instance by UUID and/or displayname,
+// and describes a command to execute for this instance.
+message AppCommand {
+   // Reference the application instance by UUID.
+   // At least one of the UUID and displayname should be defined.
+   string uuid = 1;
+   // Reference the application instance by the user-friendly displayname.
+   // At least one of the UUID and displayname should be defined.
+   string displayname = 2;
+   // Timestamp to record when the request to run the command was made.
+   // The format of the timestamp is not defined. It can be a Unix timestamp
+   // or a different time representation. It is not even required for the timestamp
+   // to match the real time or to be in-sync with the device clock.
+   // What is required, however, is that two successive but distinct requests made
+   // for the same application and with the same command will have different
+   // timestamps attached. This requirement applies even between restarts
+   // of the Local profile server. A request made after a restart should not have
+   // the same timestamp attached as the previous request with the same arguments
+   // made before the restart.
+   //
+   // EVE guarantees that a newly added command request or a change of the timestamp
+   // will result in the command being triggered ASAP, unless the Local Profile
+   // Server restarted or the device rebooted shortly after the request was made,
+   // in which case it could get lost before EVE could see/apply it.
+   uint64 timestamp = 3;
+   enum Command {
+      COMMAND_UNSPECIFIED = 0;
+      COMMAND_RESTART = 1;
+      COMMAND_PURGE = 2;
```

Comment:
The purge operation applies to some set of volumes. An application can have some set of implicitly created volumes, each with a "purge" flag being true or false, and also be attached to some persistent volumes.

We need to define exactly the scope if purge with those volumes in mind. I think we have other use cases where folks want to purge a persistent data volume to clear out any cached from a previous use of the app instance but the use case for what you are presenting seems to be to fix problems by reverting e.g., the operating system/boot volume back to a known golden image.

Reply:
This is the only info for purge I could find in the documentation:
```
Purge an ECO

An ECO which is either running or transitioning to running state will be stopped and the mutated run time state of the ECO is deleted. A subsequent action to start the ECO will start the ECO with a pristine runtime state.
```
My understanding is that it relates to this: https://github.com/lf-edge/eve/blob/master/pkg/pillar/types/zedmanagertypes.go#L62 (for which the customer wants to have a local trigger).
But maybe I could redesign the API to allow to narrow the purge down to a single volume if you see this as being useful (and will not have cases where it could be in conflict with config from controller).

Reply:
I don't think we need to generalize it here and now but it makes sense to document (in a comment in the proto file?)  that the local purge command will cause the same action as the purge command from the controller.

Reply:
Added explanation to the proto file.